packages/worker/src/worker.ts

```diff
@@ -140,6 +140,33 @@ export type ActivityTaskWithContext = ContextAware<{
 
 type CompiledWorkerOptionsWithBuildId = CompiledWorkerOptions & { buildId: string };
 
+/**
+ * Combined error information for {@link Worker.runUntil}
+ */
+export interface CombinedWorkerRunErrorCause {
+  /**
+   * Error thrown by a Worker
+   */
+  workerError: unknown;
+  /**
+   * Error thrown by the wrapped promise or function
+   */
+  innerError: unknown;
+}
+
+/**
+ * Error thrown by {@link Worker.runUntil}
+ */
+export class CombinedWorkerRunError extends Error {
+  public readonly name = 'CombinedWorkerRunError';
+  public readonly cause: CombinedWorkerRunErrorCause;
```

Comment:
`Error.cause` is a pretty common property, and most devs expect this one to be an error, not an array. It might be safer in our case to name that property `causes` (plural).

Reply:
According to MDN, you can put whatever you want in `cause`.